src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -447,6 +447,8 @@ The .NET Foundation licenses this file to you under the MIT license.
       <CustomLinkerArg Include="-s ERROR_ON_UNDEFINED_SYMBOLS=0" />
       <CustomLinkerArg Condition="'$(NativeDebugSymbols)' != 'true'" Include="-O2 -flto" />
       <CustomLinkerArg Condition="'$(NativeDebugSymbols)' == 'true'" Include="-g3" />
+      <CustomLinkerArg Condition="'$(ExportsFile)' != ''" Include="-s EXPORTED_FUNCTIONS=@&quot;$(ExportsFile)&quot;" />
+      <CustomLinkerArg Condition="$(NativeLib) == 'Shared'" Include="-s EXPORTED_FUNCTIONS=_NativeAOT_StaticInitialization" />
```

Comment:
I just tested this locally, it appears `emcc` takes the last `EXPORTED_FUNCTIONS` argument as the definitive one, i. e. `_NativeAOT_StaticInitialization` overrides the exports file list.

We could modify the file in-place to include to include `_NativeAOT_StaticInitialization`, but ideally we wouldn't need that at all and it would be called automatically. Let me drill in to see why this is not happening.

Reply:
> Hmm, I've tested this separately before submitting and I saw both symbols exported.
> 
> >

Weird, you're right.

Reply:
> We could modify the file in-place to include to include `_NativeAOT_StaticInitialization`, but ideally we wouldn't need that at all and it would be called automatically. Let me drill in to see why this is not happening.

From what I saw, that function is defined in C, right? If so, adding `__attribute__((used))` should be in theory enough if it's also linked through the normal wasm-ld process.

Reply:
> If so, adding `__attribute__((used))` should be in theory enough if it's also linked through the normal wasm-ld process.

Agree. I wonder why other targets don't use this mechanism.

I think for this change it should be ok to only touch the linker part and add it with `<CustomLinkerArg Condition="$(NativeLib) == 'Shared'" Include="-Wl,--export,NativeAOT_StaticInitialization" />`.

Reply:
> Agree. I wonder why other targets don't use this mechanism.

Yeah as I mentioned on the original issue, ideally I'd use it for all targets for `UnmanagedCallersOnly` functions as well instead of intermediate exports file, but for now just trying to match existing behaviour of other platforms 🤷‍♂️

I do think the `used`-based mechanism would be somewhat easier to maintain though.

Reply:
> I think for this change it should be ok to only touch the linker part and add it with `<CustomLinkerArg Condition="$(NativeLib) == 'Shared'" Include="-Wl,--export,NativeAOT_StaticInitialization" />`.

Ah neat, wasn't sure if that would work.

Reply:
I think it still needs to be `_`-prefixed like `--export,_NativeAOT_StaticInitialization` though?